### PR TITLE
Implementing Polaris-style scannables and cataloguer.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes

Adds a cataloguer that can be used to scan exoflora/plants to unlock codex pages and earn points. Opening this primarily to add it to the project tracker before I forget the branch exists.

**TODO:**
- [x] Implement scan beam.
- [X] Implement scan range indicator.
- ~~Implement some basic pages for exoflora/exofauna.~~
- [x] Add fabricator recipe for cataloguer.

## Why and what will this PR improve

Works towards Polaris feature parity and adds a reward system for exploration.

## Authorship
Polaris for the original feature and icons, but this implementation is fresh.

## Changelog
Nothing player-facing initially.

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
Depends on #2557.

**Comment by MistakeNot4892** [comment]
I've tested this with some placeholder codex stuff locally but as currently committed there are no scannables ingame. Code is otherwise ready for review though.

File: code/_helpers/visual_filters.dm [hunk 1]
```diff
@@ -26,8 +29,9 @@
 	UPDATE_OO_IF_PRESENT
 
 /atom/movable/proc/get_filter(filter_name)
-	if(filter_data && filter_data[filter_name])
-		return filters[filter_data.Find(filter_name)]
+	var/filter_index = filter_data?.Find(filter_name)
+	if(filter_index > 0 && filter_index <= length(filters))
```
Comment (MistakeNot4892) [inline code/_helpers/visual_filters.dm:33]: I was getting a bunch of runtimes with filters that I'm not sure how to debug or explain. This check at least stopped it runtiming in the middle of unrelated code.

File: code/game/objects/compass/compass_holder.dm [hunk 2]
```diff
@@ -54,7 +54,7 @@ var/global/list/angle_step_to_dir = list(
 		M.Translate(0, get_label_offset())
 		M.Turn(effective_compass_period * i)
 		I.transform = M
-		I.filters = filter(type="drop_shadow", color = "#77777777", size = 2, offset = 1,x = 0, y = 0)
+		I.add_filter("glow", 1, list("drop_shadow", color = "#77777777", size = 2, offset = 1,x = 0, y = 0))
```
Comment (MistakeNot4892) [inline code/game/objects/compass/compass_holder.dm:57]: Fixed these up while trying to hunt down filter issues mentioned earlier.

File: code/modules/codex/codex_cataloguer.dm [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+			loaded_disk = W
+			playsound(user.loc, 'sound/weapons/flipblade.ogg', 50, 1)
+			to_chat(user, SPAN_NOTICE("You slot \the [W] into \the [src]."))
+		return TRUE
+	return ..()
+
+/obj/item/cataloguer/examine(mob/user, distance, infix, suffix)
+	. = ..()
+	if(loaded_disk)
+		to_chat(user, "It has \a [loaded_disk] slotted into the storage port. The display indicates it currently holds [loaded_disk.data] Good Explorer Point\s.")
```
Comment (MistakeNot4892) [inline code/modules/codex/codex_cataloguer.dm:199]: Willing to change this away from a GBP reference, just using the same language as the pre-existing explo point scheme.

**Comment by MistakeNot4892** [comment]
This PR implements the framework for scannables but doesn't actually implement anything that uses the system - as scannables will generally be entirely lore related I'm comfortable leaving it as is.

**Review by afterthought2 — APPROVED** [review]
